### PR TITLE
Remove BaseAssemblySpec::HasUniqueIdentity (always returns true)

### PR DESCRIPTION
**Pull request by elinor-fung**
`BaseAssemblySpec::HasUniqueIdentity` is a remnant from built-in WinRT support. It always returns true now that built-in WinRT support is gone.

Also moved some assembly spec files out of DAC compile that don't need to be there.

cc @AaronRobinsonMSFT @jkoritzinsky @vitek-karas

### CODE REVIEW

File: src/coreclr/vm/appdomain.cpp [hunk 1]
```diff
@@ -3096,10 +3096,10 @@ DomainAssembly *AppDomain::LoadDomainAssemblyInternal(AssemblySpec* pIdentity,
     {
         AssemblySpec spec;
         spec.InitializeSpec(result->GetFile());
-        if (spec.CanUseWithBindingCache() && result->CanUseWithBindingCache())
+        if (result->CanUseWithBindingCache())
```
Comment (jkotas) [inline src/coreclr/vm/appdomain.cpp:3099]: Do these other `CanUseWithBindingCache` methods always return true  as well?

Comment (elinor-fung) [reply src/coreclr/vm/appdomain.cpp:3099]: Indeed, they do.
`PEAssembly::CanUseWithBindingCache` - and thus `DomainAssembly::CanUseWithBindingCache` - always returns true. (`PEFile::CanUseWithBindingCache` does not appear to be used.)

**Review by jkotas — APPROVED** [review]
Thank you!

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Review by vitek-karas — APPROVED** [review]
:shipit:

**Pull request merged**